toplevel/vernacentries.ml

```diff
@@ -136,7 +136,10 @@ let make_cases_aux glob_ref =
 		   let n' = Namegen.next_name_away_with_default (Id.to_string Namegen.default_dependent_ident) n avoid in
 		   Id.to_string n' :: rename (n'::avoid) l in
 	     let al' = rename [] al in
-	     (Id.to_string consname :: al') :: l)
+	     let dirpath = Nametab.dirpath_of_global glob_ref in
+	     let consname_qualid = Libnames.make_qualid dirpath consname in
+	     let consname_global = Nametab.locate consname_qualid in
+	     (Libnames.string_of_qualid (Nametab.shortest_qualid_of_global Id.Set.empty consname_global) :: al') :: l)
```

Comment:
Should probably be ok. The `Id.Set.t` is to avoid some extra variable names (e.g in a proof, to avoid the hypotheses), but here, there are only constructors to disambiguate, and these cannot be hidden by variables.